docs/versioned_docs/version-2.11.0/troubleshooting/troubleshooting.md

```diff
@@ -0,0 +1,189 @@
+
+2019-02-05/15:43:30.783 worker            :6292: c#1.140005666154240.: worker 1/-1/43088 end (c#1.140005666154240/s#1.140005649413888)
+2019-02-05/15:43:33.285 worker            :5885: c#2.140005649413888.: con:2/18 [8d614b93:43092->10.233.45.100:3260,1]
+2019-02-05/15:43:33.536 istgt_iscsi_op_log:1923: c#2.140005649413888.: login failed, target not ready
+
+2019-02-05/15:43:33.787 worker            :6088: c#2.140005649413888.: iscsi_read_pdu() EOF
+
+2019-02-05/15:43:33.787 sender            :5852: s#2.140005632636672.: sender loop ended (2:18:43092)
+
+2019-02-05/15:43:33.788 worker            :6292: c#2.140005649413888.: worker 2/-1/43092 end (c#2.140005649413888/s#2.140005632636672)
+2019-02-05/15:43:35.251 istgt_remove_conn :7039: c#0.140005771040512.: remove_conn->initiator:147.75.97.141(iqn.2019-02.net.packet:device.7c8ad781) Target: 10.233.109.82(dummy LU0) conn:0x7f55a4c18000:0 tsih:1 connections:0  IOPending=0
+2019-02-05/15:43:36.291 worker            :5885: c#0.140005666154240.: con:0/14 [8d614b93:43094->10.233.45.100:3260,1]
+2019-02-05/15:43:36.540 istgt_iscsi_op_log:1923: c#0.140005666154240.: login failed, target not ready
+```
+
+**Troubleshooting**
+
+The cause of high memory consumption of kubelet is mainly due to the following.
+
+There are 3 modules are involved - cstor-istgt, kubelet and iscsiInitiator(iscsiadm). kubelet runs iscsiadm command to do discovery on cstor-istgt. If there is any delay in receiving response of discovery opcode (either due to network or delay in processing on target side), iscsiadm retries few times, and, gets into infinite loop dumping error messages as below:
+
+```shell hideCopy
+iscsiadm: Connection to Discovery Address 127.0.0.1 failed
+iscsiadm: failed to send SendTargets PDU
+iscsiadm: connection login retries (reopen_max) 5 exceeded
+iscsiadm: Connection to Discovery Address 127.0.0.1 failed
+iscsiadm: failed to send SendTargets PDU
+```
+
+kubelet keeps taking this response and accumulates the memory. More details can be seen [here](https://github.com/openebs/openebs/issues/2382).
+
+**Workaround**
+
+Restart the corresponding istgt pod to avoid memory consumption.
+
+## Application and OpenEBS pods terminate/restart under heavy I/O load {#Pods-restart-terminate-when-heavy-load}
+
+This is caused due to lack of resources on the Kubernetes nodes, which causes the pods to evict under loaded conditions as the node becomes _unresponsive_. The pods transition from _Running_ state to _unknown_ state followed by _Terminating_ before restarting again.
+
+**Troubleshooting**
+
+The above cause can be confirmed from the `kubectl describe pod` which displays the termination reason as _NodeControllerEviction_. You can get more information from the kube-controller-manager.log on the Kubernetes master.
+
+**Workaround:**
+
+You can resolve this issue by upgrading the Kubernetes cluster infrastructure resources (Memory, CPU).
+
+<font size="6" color="orange">Others</font>
```

Comment:
Use markdown heading. Please change this in other places as well

Reply:
In markdown, we cannot color the heading using `###`. For that HTML tag will be used, so retained the previous code